simulation-construction-set/falling-brick/src/main/java/us/ihmc/fallingBrick/FallingBrickRobot.java

```diff
@@ -0,0 +1,204 @@
+       * associated with the joint. The second parameter "new Vector3d(0.0, 0.0, 0.0)" defines the offset
+       * of this joint from the previous joint. The third parameter "this" refers to the robot itself.
+       */
+      floatingJoint = new FloatingJoint("base", new Vector3D(0.0, 0.0, 0.0), this);
+      Link link1 = base("base", YoAppearance.Purple());
+      // Sets the link created previously as the link for this joint
+      floatingJoint.setLink(link1);
+
+      // Add a RootJoint to the robot
+      /*
+       * This line adds floatingJoint as a rootJoint of the robot. In order to be part of a robot, a joint
+       * must either be added as a root joint, or be attached to a parent joint. This ensures the tree
+       * structure (forest structure if there are multiple root joints) of the robot.
+       */
+      addRootJoint(floatingJoint);
+
+      // add ground contact points to the brick
+      GroundContactPoint gc1 = new GroundContactPoint("gc1", new Vector3D(BASE_L / 2.0, BASE_W / 2.0, BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc1);
+      GroundContactPoint gc2 = new GroundContactPoint("gc2", new Vector3D(BASE_L / 2.0, -BASE_W / 2.0, BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc2);
+      GroundContactPoint gc3 = new GroundContactPoint("gc3", new Vector3D(-BASE_L / 2.0, BASE_W / 2.0, BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc3);
+      GroundContactPoint gc4 = new GroundContactPoint("gc4", new Vector3D(-BASE_L / 2.0, -BASE_W / 2.0, BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc4);
+
+      GroundContactPoint gc5 = new GroundContactPoint("gc5", new Vector3D(BASE_L / 2.0, BASE_W / 2.0, -BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc5);
+      GroundContactPoint gc6 = new GroundContactPoint("gc6", new Vector3D(BASE_L / 2.0, -BASE_W / 2.0, -BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc6);
+      GroundContactPoint gc7 = new GroundContactPoint("gc7", new Vector3D(-BASE_L / 2.0, BASE_W / 2.0, -BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc7);
+      GroundContactPoint gc8 = new GroundContactPoint("gc8", new Vector3D(-BASE_L / 2.0, -BASE_W / 2.0, -BASE_H / 2.0), this);
+      floatingJoint.addGroundContactPoint(gc8);
+
+      GroundContactPoint gc9 = new GroundContactPoint("gc9", new Vector3D(0.0, 0.0, BASE_H / 2.0 + BASE_H), this);
+      floatingJoint.addGroundContactPoint(gc9);
+      GroundContactPoint gc10 = new GroundContactPoint("gc10", new Vector3D(0.0, 0.0, -BASE_H / 2.0 - BASE_H), this);
+      floatingJoint.addGroundContactPoint(gc10);
+
+      this.setController(this); // tells the simulator to call the local doControl() method
+
+      // instantiate ground contact model
+      GroundContactModel groundModel = new LinearGroundContactModel(this, 1422, 150.6, 50.0, 1000.0, getRobotsYoVariableRegistry());
+      GroundProfile3D profile = new WavyGroundProfile();
```

Comment:
Maybe of piece comment for that guy too:
A `GroundProfile3D` is a height map that is used to define the ground.